libseccomp/src/filter_context.rs

```diff
@@ -99,6 +99,34 @@ impl ScmpFilterContext {
         Ok(())
     }
 
+    /// Precompute the seccomp filter for future use
+    ///
+    /// This function precomputes the seccomp filter and stores it internally for
+    //  future use, speeding up seccomp_load() and other functions which require
+    //  the generated filter.
+    ///
+    /// This function corresponds to
+    /// [`seccomp_merge`](https://man7.org/linux/man-pages/man3/seccomp_precompute.3.html).
+    ///
+    /// # Errors
+    ///
+    /// If precomputing the filter fails, an error will be returned.
+    ///
+    /// # Examples
+    /// ```
+    /// # use libseccomp::*;
+    /// let mut ctx = ScmpFilterContext::new_filter(ScmpAction::Allow)?;
+    /// ctx.precompute()?;
+    /// let syscall = ScmpSyscall::from_name("dup3")?;
+    /// ctx.add_rule(ScmpAction::KillThread, syscall)?;
+    /// ctx.precompute()?;
+    /// ctx.load()?;
+    /// # Ok::<(), Box<dyn std::error::Error>>(())
+    /// ```
+    pub fn precompute(&self) -> Result<()> {
+        cvt(unsafe { seccomp_precompute(self.ctx.as_ptr()) })
```

Comment:
Certainly it has to be behind a `cfg(libseccomp_v2_6)`. However, we have to decide where we put it hand how we set it. See discussions in #217 and #121.